### PR TITLE
array methods TDD tests

### PR DESCRIPTION
**Pull request by christylaguardia**

### CODE REVIEW

**Comment by martypdx** [comment]
@christylaguardia major part of this assignment was to write your own implementation of array methods, not wrap the built-in JavaScript array methods.

Near top of `LAB.md`:

> Implement select array methods from scratch. Only use:
> * `array.length`
> * `array[i] = value`
> * `value = array[i]`

File: src/array-methods.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const methods = {};
+// push
+methods.push = function(array, item) {
+    array.push(item);
+    return array.length;
+};
+// for each
+methods.foreach = function(array) {
```
Comment (nsmeds) [inline src/array-methods.js]: You'll need to follow the function signature specified in [LAB.md](https://github.com/cfpdx-401JS-summer-2017/array-methods/blob/master/LAB.md) and pass a callback along with the array. Currently your function only takes a single parameter, `array`. This should simplify the rest of your function.

File: src/array-methods.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        if (Number.isInteger(array[i])) {
+            numbers.push(array[i]);
+        }
+    }
+    // console.log('count of numbers in array', numbers.length);
+    return numbers.length;
+};
+
+// map
+methods.map = function(array) {
```
Comment (nsmeds) [inline src/array-methods.js]: Same as above: should be `function(array, callback)`

File: src/array-methods.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const newArray = [];
+    for (var i = 0; i < array.length; i++) {
+        newArray[i] = array[i];
+    }
+    // console.log(newArray);
+    return newArray.length;
```
Comment (nsmeds) [inline src/array-methods.js]: I think you are getting too specific by returning `newArray.length` - your function should just return newArray. If you specifically need to deal with the length, you can do that in your test.

File: src/array-methods.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+};
+// filter
+methods.filter = function(array) {
+    const filteredArray = array.filter(function(item) {
+        return Number.isInteger(item);
+    });
+    return filteredArray.length;
```
Comment (nsmeds) [inline src/array-methods.js]: Same as above re: `array, callback` and `length`.

**Pull request merged**